boards/BMS/Test/Inc/Test_Imd.h

```diff
@@ -15,7 +15,7 @@ class ImdTest : public testing::Test
         enum Imd_ConditionName condition_name,
         float &                fake_pwm_frequency_return_val);
     static void
-        SetPwmFrequencyTolerance(struct Imd *imd_to_set, float tolerance);
+        SetPwmFrequencyTolerance(struct Imd *&imd_to_set, float tolerance);
```

Comment:
Ok, ya got me, why are we changing this? :thinking:

Reply:
I was a bit confused at first too. So the problem is that the new pointer returned by `_Create` in this function is stored in the local variable `imd_to_set` which gets lost after this function returns. 
```
void ImdTest::SetPwmFrequencyTolerance(struct Imd *&imd_to_set, float tolerance)
{
    TearDownObject(imd_to_set, App_Imd_Destroy);
    imd_to_set = App_Imd_Create(
        get_pwm_frequency, tolerance, get_pwm_duty_cycle,
        get_seconds_since_power_on);
}
```